src/quest_manager/models.py

```diff
@@ -975,3 +989,142 @@ def get_minutes_to_complete(self):
+        )
+    )
+    solution_text = models.TextField(
+        null=True,
+        blank=True,
+        help_text=(
+            "This text will appear to the marker, allowing for an easy comparison of "
+            "the student's submission to an ideal solution."
+        ),
+    )
+    solution_file = RestrictedFileField(
+        blank=True,
+        null=True,
+        upload_to="quest/question/solution/%Y/%m/%d",
+        help_text=(
+            "This file will appear to the marker, allowing them to easily compare "
+            "the student's submission to the ideal solution for file-based questions."
+        ),
+    )
+
+    # Each tuple contains the file type as both the stored value and the display value
+    # e.g. ("media", "Media") or ("image", "Image")
+    allowed_file_type = models.CharField(
+        max_length=50,
+        choices=[(key, key.capitalize()) for key in FILE_MIME_TYPES.keys()],
+        default="media",
+        help_text=("The types of files that can be uploaded by a student for this question.")
+    )
+
+    marker_notes = models.TextField(
+        null=True,
+        blank=True,
+        help_text=(
+            "This text will appear to the marker, providing additional details about "
+            "what a student's submission should be or how the question should be marked."
+        ),
+    )
+    required = models.BooleanField(
+        default=True,
+        help_text=(
+            "If this is checked, the student must complete this "
+            "question to complete the quest."
+        ),
+    )
+    datetime_created = models.DateTimeField(auto_now_add=True)
+    datetime_last_edit = models.DateTimeField(auto_now=True)
+    ordinal = models.PositiveIntegerField(
+        default=1,
+        help_text=(
+            "Indicates the order of the question in the quest so that "
+            "questions are displayed to the student in the correct order."
+        ),
+    )
+
+    quest = models.ForeignKey(Quest, on_delete=models.CASCADE)
+    type = models.CharField(
+        max_length=20, choices=QuestionType.choices, default=QuestionType.SHORT_ANSWER
+    )
+
+    class Meta:
+        ordering = ["ordinal"]
+        # ensure two questions in the same quest cannot have the same ordinal
+        unique_together = ["quest", "ordinal"]
+
+    def __str__(self):
+        """Return string representation in format of: [Quest Name] ([Question Type] - [Question Ordinal])
+        e.g. "Quest 1 (short_answer - 1)" for the first question in the quest with name "Quest 1"
+        """
+        return f"{self.quest.name} ({self.type} - {self.ordinal})"
+
+
+class QuestionSubmission(models.Model):
+    """Used for storing a student's submission for a question.
+
+    This model is used for all types of questions, including multiple choice, short answer, file upload, etc.
+    When a student submits a quest, each Question will have a QuestionSubmission created for it.
+    """
+
+    response_text = models.TextField(
```

Comment:
Is this intended for short answer questions?  Short answer should be a CharField with 128 characters or so.  A TextField would be displayed with a Summernote widget for long-answer (with formatting and embedding etc)

Reply:
I was under the impression that we'd use the same for both, and that for short answer, we'd limit the length in the form field. How does that sound? I think it would be simpler than adding  additional `short_solution_text` and `short_response_text` fields.

Reply:
Yep, that makes sense. Short answer can use a charfield widget on the form, and Long answer can use a SafeSummernote. Might need to also look at their processing and display (safe filter or not).

Good idea!